### PR TITLE
CI: chain requirements to ensure compatibility

### PR DESCRIPTION
**Pull request by Borda**
*Issue #, if available:* atm, all install requirements are installed as stand-alone, which means if there is a version collision, `pip` does not care about it, and so for eventual version compatibility, a collision may not be discovered

*Description of changes:*
just install all needed requirements for the particular job at once so pip sees all required packages and finds an intersection among all :flags: 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

cc: @jaheba @kashif @lostella kind ling as it seems there are no revivers by default :chipmunk: 

**Please tag this pr with at least one of these labels to make our release process faster:** BREAKING, new feature, bug fix, other change, dev setup

### CODE REVIEW

**Comment by lostella** [comment]
@Borda thanks! I guess let's wait for all checks to be green before coming back to this

**Review by lostella — DISMISSED** [review]
Thanks @Borda! Just needs rebasing

**Comment by Borda** [comment]
> Just needs rebasing

@lostella done; just curious why rebasing and not just merging the `dev` branch?

**Comment by lostella** [comment]
> > Just needs rebasing
> 
> @lostella done; just curious why rebasing and not just merging the `dev` branch?

whatever works :-) what I meant is to update the branch so that CI workflows run

File: requirements/requirements-rotbaum-extra-methods.txt [hunk 1]
```diff
@@ -1,2 +1,2 @@
 lightgbm>=2.3.1
-scikit-garden>=0.1.2
+scikit-garden @ https://github.com/scikit-garden/scikit-garden/archive/refs/heads/master.zip
```
Comment (Borda) [inline requirements/requirements-rotbaum-extra-methods.txt]: the last published version is from 2017 but the repo has some updates from 2021

**Comment by Borda** [comment]
If possible, I would suggest dropping `scikit-garden` as it has been a stale project since 2017, and there seems to be no such intention for updating at least to installable without problems, see https://github.com/scikit-garden/scikit-garden/issues/105#issuecomment-1923509842

**Comment by kashif** [comment]
we can also move the pytorch version to 2.x no?

**Comment by Borda** [comment]
> we can also move the pytorch version to 2.x no?

probably but lets make it as separate PR :chicken:

File: .github/workflows/pypi.yml [hunk 2]
```diff
@@ -16,10 +16,6 @@ jobs:
       uses: actions/setup-python@v4
       with:
         python-version: '3.8'
-    - name: Install dependencies
-      run: |
-        python -m pip install -U pip
-        python -m pip install setuptools wheel
```
Comment (lostella) [inline .github/workflows/pypi.yml]: I’m not sure about the changes here, can we revert these?

**Comment by lostella** [comment]
> If possible, I would suggest dropping `scikit-garden` as it has been a stale project since 2017, and there seems to be no such intention for updating at least to installable without problems, see [scikit-garden/scikit-garden#105 (comment)](https://github.com/scikit-garden/scikit-garden/issues/105#issuecomment-1923509842)

Yeah we should get rid of that

File: .github/workflows/pypi.yml [hunk 3]
```diff
@@ -16,10 +16,6 @@ jobs:
...
     - name: Create `dist` folder
```
Comment (Borda) [inline .github/workflows/pypi.yml]:
```suggestion
    - name: Install dependencies
      run: |
        python -m pip install -U pip
        python -m pip install setuptools wheel
    - name: Create `dist` folder
```

File: .github/workflows/pypi.yml (hunk 2 above)
Comment (Borda) [reply .github/workflows/pypi.yml]: done

**Review by lostella — APPROVED** [review]
🚀

**Pull request merged**